lib/rspec-puppet/facter_impl.rb

```diff
@@ -8,7 +8,13 @@ def initialize
     end
 
     def value(fact_name)
-      @facts[fact_name.to_s]
+      fact_name.to_s.split('.').reduce(@facts) do |sub_hash, key|
+        if sub_hash.is_a?(Hash) && sub_hash.key?(key)
+          sub_hash[key]
+        else
+          return nil
+        end
+      end
```

Comment:
Any reason you choose this over `.dig`? I think this does the same:
```suggestion
      facts.dig(*fact.to_s.split('.'))
```

```
[1] pry(main)> fact = 'os.family'
=> "os.family"
[2] pry(main)> facts = { 'os' => { 'family' => 'RedHat' } }
=> {"os"=>{"family"=>"RedHat"}}
[3] pry(main)> facts.dig(*fact.to_s.split('.'))
=> "RedHat"
[4] pry(main)> facts.dig(*'os.name'.split('.'))
=> nil
```

It does fail if a key is a string where it expects a hash:
```
[5] pry(main)> facts.dig(*'os.family.bla'.split('.'))
TypeError: String does not have #dig method
from (pry):5:in `dig'
```

So you would need to catch that.